api/views/uqj.py

```diff
@@ -37,3 +41,15 @@ def get_queryset(self):
                 accessible_uqj.append(temp.question)
         uqj = user.question_junctions.filter(question__in=accessible_uqj)
         return uqj
+
+    @action(detail=True, methods=['post'], url_path='switch-favorite')
+    def switch_favorite(self, request, pk=None):
```

Comment:
For the convention of the project, we don't have toggle APIs. We have create/update APIs. And also this is a read-only api, you should not update anything here.
I think we don't have a viewset to update UQJs. You should create a viewset to update UQJ, which only allows a user to update is_favorite ONLY for the their UQJs.